src/Components/Server/src/CircuitDisconnectMiddleware.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.Server.Circuits;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Components.Server
+{
+    internal class CircuitDisconnectMiddleware
+    {
+        private const string CircuitIdKey = "circuitId";
+
+        public CircuitDisconnectMiddleware(
+            ILogger<CircuitDisconnectMiddleware> logger,
+            CircuitRegistry registry,
+            CircuitIdFactory circuitIdFactory,
+            RequestDelegate next)
+        {
+            Logger = logger;
+            Registry = registry;
+            CircuitIdFactory = circuitIdFactory;
+            Next = next;
+        }
+
+        public ILogger<CircuitDisconnectMiddleware> Logger { get; }
+        public CircuitRegistry Registry { get; }
+        public CircuitIdFactory CircuitIdFactory { get; }
+        public RequestDelegate Next { get; }
+
+        public async Task Invoke(HttpContext context)
+        {
+            if (!HttpMethods.IsPost(context.Request.Method))
+            {
+                context.Response.StatusCode = StatusCodes.Status405MethodNotAllowed;
+                return;
+            }
+
+            var form = await context.Request.ReadFormAsync();
```

Comment:
it would be a good idea to validate the content type here or else this will throw.

Reply:
Also, the use of form-encoded data here is quite unusual, but I like it. Simpler than having to have a type to JSON-deserialize to 👍

Reply:
We send a FormData object on the client. The send beacon API doesn't give you control over the request, (AFAIK). We could have passed a string, but that likely arrived as text/plain.

I think using FormData is a better choice for this particular case. I'll add a check here for the content type.